src/t_string.c

```diff
@@ -411,12 +411,9 @@ void getexCommand(client *c) {
 
 void getdelCommand(client *c) {
     if (getGenericCommand(c) == C_ERR) return;
-    int deleted = server.lazyfree_lazy_user_del ? dbAsyncDelete(c->db, c->argv[1]) :
-                  dbSyncDelete(c->db, c->argv[1]);
-    if (deleted) {
+    if (dbSyncDelete(c->db, c->argv[1])) {
         /* Propagate as DEL/UNLINK command */
-        robj *aux = server.lazyfree_lazy_user_del ? shared.unlink : shared.del;
-        rewriteClientCommandVector(c,2,aux,c->argv[1]);
+        rewriteClientCommandVector(c,2,shared.del,c->argv[1]);
```

Comment:
Suggestion: Change the comment too.

Reply:
Good catch - updated in 8dd1a910b03c80700f85307b38914103ba08d5b1!